### ISSUE TITLE
Resetting a timeseries value periodically

### ISSUE THREAD

**Issue by inanlia** [thingsboard/thingsboard#10506]
* Rule Engine

**Description**
What I'm trying to do is to calculate the working hours of my machine. I do this by substracting the old timestamp from the new one and adding the difference to another value. I also put a check so that when the device stops working the time between does not count.

But when I wamt to show on a graph daily working hours, what I came up with is getting all the data from a certain interval, divide them into groups and substract the first value from the last. While this works, the calculation time increases with the amount of data.

One of the solutions I came up with is to keep to total amount of working hours as a attribute and reset the daily working hour at midnight. This way when I want to show the daily values I can aggregate the data to show only the MAX value.

So how can I reset the value at midnight? 

**Environment**
 * ThingsBoard: 3.6.2

**Comment by inanlia** [comment]
I believe one way of doing this using Generator Code. 
I would have to add it at the midnight give it 86.400 seconds as a period. 
I'm still open to suggestions.

I also tried to get the MIN and MAX values for each day but to do that I had to use custom subscription.
My problem was here #10496. 
Aggregation kept defaulting to AVG and I also found out that when data is updated the widget used 90% of the cpu so that is not a good way of doing this.

**Comment by mykhailokornieiev** [comment]
Hi
You can use scheduler instead of generator, so it will update attribute that will trigger necessary rule chain.
If I understand your request correctly, then there is no standard solution to “reset” aggregation in aggregate stream rule node, so it will delete already aggregated value (250 for example) and start it over (from 0)
Workarounds:
1. Delete all related data - aggregated keys, persistence (RuleNodeState_ID) and it’s telemetry data
2. Recreate aggregate stream rule node(s)

**Comment by inanlia** [comment]
Hi,

The generator node did not work correctly. I added it near midnight hoping that it would work every night to add a single 0 value to database so that my calculations would work correctly. Unfortunately the generator node started to create a 0 value at 10 am.

Instead of creating a new 0 value, this time while getting previous value, I also added the value's timestamp into the data. This way I could compare the new data ts and old data ts and if they are different days I started to count from zero.
